cub/cub/block/block_load.cuh

```diff
@@ -206,22 +206,33 @@ InternalLoadDirectBlockedVectorized(int linear_tid, const T* block_src_ptr, T (&
   _CCCL_DIAG_POP
   constexpr int vector_size        = (total_words % 4 == 0) ? 4 : (total_words % 2 == 0) ? 2 : 1;
   constexpr int vectors_per_thread = total_words / vector_size;
-  using vector_t                   = typename CubVector<device_word_t, vector_size>::Type;
 
-  // Load into an array of vectors in thread-blocked order
-  vector_t vec_items[vectors_per_thread];
-  const vector_t* vec_ptr = reinterpret_cast<const vector_t*>(block_src_ptr) + linear_tid * vectors_per_thread;
-#  pragma unroll
-  for (int i = 0; i < vectors_per_thread; i++)
+  // Add the alignment check to ensure the vectorized loading can proceed.
+  if (reinterpret_cast<uintptr_t>(block_src_ptr) % (vector_size << 2) == 0)
```

Comment:
Q: I don't understand this test. Shouldn't this test the address to be a multiple of `sizeof(vector_t)` (which should be `sizeof(T) * vector_size`)